clouddriver-kubernetes/src/test/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/caching/agent/KubernetesCacheDataConvertSpec.groovy

```diff
@@ -129,4 +132,45 @@ metadata:
     KubernetesKind.SERVICE     | KubernetesApiVersion.V1           | ["cluster": [Keys.cluster("account", "name")], "application": [Keys.application("blarg")]]
     KubernetesKind.SERVICE     | KubernetesApiVersion.V1           | ["cluster": [Keys.cluster("account", "name")], "application": [Keys.application("blarg"), Keys.application("asdfasdf")]]
   }
+
+  def filterRelationships(Collection<String> keys, List<Triple<KubernetesApiVersion, KubernetesKind, String>> existingResources) {
+    return keys.findAll { sk ->
+      def key = (Keys.InfrastructureCacheKey) Keys.parseKey(sk).get()
+      return existingResources.find { Triple<KubernetesApiVersion, KubernetesKind, String> lb ->
+        return lb.getLeft() == key.getKubernetesApiVersion() && lb.getMiddle() == key.getKubernetesKind() && lb.getRight() == key.getName()
+      } != null
```

Comment:
Seems more like an `any` rather than a `find`.

Reply:
I tried `any` at first and it never iterated over any elements in the list... not sure what's wrong there.